cmd/peer-rest-client.go

```diff
@@ -676,6 +676,32 @@ func (client *peerRESTClient) BackgroundHealStatus() (madmin.BgHealState, error)
 	return state, err
 }
 
+func (client *peerRESTClient) getLocalDiskIDsWith() chan []string {
+	ch := make(chan []string, 1)
+	go func() {
+		respBody, err := client.call(peerRESTMethodGetLocalDiskIDs, nil, nil, -1)
+		if err != nil {
+			ch <- nil
+		}
+		defer http.DrainBody(respBody)
+		var diskIDs []string
+		gob.NewDecoder(respBody).Decode(&diskIDs)
+		ch <- diskIDs
+	}()
+	return ch
+}
+
+// GetLocalDiskIDs - get a peer's local disks' IDs.
+func (client *peerRESTClient) GetLocalDiskIDs() []string {
+	ch := client.getLocalDiskIDsWith()
+	select {
+	case s := <-ch:
+		return s
+	case <-time.After(globalReadyDeadline):
```

Comment:
you can log here saying that deadline exceeded or something? perhaps nicer would pass a context and use context.Done() to ensure that the call is canceled, it would be perhaps good to do use callWithContext() - 10secs for reach peer call rather than 10secs for entire peer call.

Reply:
callWithContext() is better, will fix